Provide compatibility with 2026.5 - deprecated warnings

```
src/esphome/components/sinclair_ac/esppac.cpp: In member function 'void esphome::sinclair_ac::SinclairAC::update_swing_horizontal(const std::string&)':
src/esphome/components/sinclair_ac/esppac.cpp:138:41: warning: 'esphome::select::Select::state' is deprecated: Use current_option() instead of .state. Will be removed in 2026.5.0 [-Wdeprecated-declarations]
  138 |         this->horizontal_swing_select_->state != this->horizontal_swing_state_)
      |                                         ^~~~~
In file included from src/esphome/components/sinclair_ac/esppac.h:5,
                 from src/esphome/components/sinclair_ac/esppac.cpp:2:
src/esphome/components/select/select.h:39:15: note: declared here
   39 |   std::string state{};
      |               ^~~~~
src/esphome/components/sinclair_ac/esppac.cpp:138:41: warning: 'esphome::select::Select::state' is deprecated: Use current_option() instead of .state. Will be removed in 2026.5.0 [-Wdeprecated-declarations]
  138 |         this->horizontal_swing_select_->state != this->horizontal_swing_state_)
      |                                         ^~~~~
src/esphome/components/select/select.h:39:15: note: declared here
   39 |   std::string state{};
      |               ^~~~~
src/esphome/components/sinclair_ac/esppac.cpp:138:41: warning: 'esphome::select::Select::state' is deprecated: Use current_option() instead of .state. Will be removed in 2026.5.0 [-Wdeprecated-declarations]
  138 |         this->horizontal_swing_select_->state != this->horizontal_swing_state_)
      |                                         ^~~~~
src/esphome/components/select/select.h:39:15: note: declared here
   39 |   std::string state{};
      |               ^~~~~
src/esphome/components/sinclair_ac/esppac.cpp: In member function 'void esphome::sinclair_ac::SinclairAC::update_swing_vertical(const std::string&)':
src/esphome/components/sinclair_ac/esppac.cpp:149:39: warning: 'esphome::select::Select::state' is deprecated: Use current_option() instead of .state. Will be removed in 2026.5.0 [-Wdeprecated-declarations]
  149 |         this->vertical_swing_select_->state != this->vertical_swing_state_)
      |                                       ^~~~~
src/esphome/components/select/select.h:39:15: note: declared here
   39 |   std::string state{};
      |               ^~~~~
src/esphome/components/sinclair_ac/esppac.cpp:149:39: warning: 'esphome::select::Select::state' is deprecated: Use current_option() instead of .state. Will be removed in 2026.5.0 [-Wdeprecated-declarations]
  149 |         this->vertical_swing_select_->state != this->vertical_swing_state_)
      |                                       ^~~~~
src/esphome/components/select/select.h:39:15: note: declared here
   39 |   std::string state{};
      |               ^~~~~
src/esphome/components/sinclair_ac/esppac.cpp:149:39: warning: 'esphome::select::Select::state' is deprecated: Use current_option() instead of .state. Will be removed in 2026.5.0 [-Wdeprecated-declarations]
  149 |         this->vertical_swing_select_->state != this->vertical_swing_state_)
      |                                       ^~~~~
src/esphome/components/select/select.h:39:15: note: declared here
   39 |   std::string state{};
      |               ^~~~~
src/esphome/components/sinclair_ac/esppac.cpp: In member function 'void esphome::sinclair_ac::SinclairAC::update_display(const std::string&)':
src/esphome/components/sinclair_ac/esppac.cpp:160:32: warning: 'esphome::select::Select::state' is deprecated: Use current_option() instead of .state. Will be removed in 2026.5.0 [-Wdeprecated-declarations]
  160 |         this->display_select_->state != this->display_state_)
      |                                ^~~~~
src/esphome/components/select/select.h:39:15: note: declared here
   39 |   std::string state{};
      |               ^~~~~
src/esphome/components/sinclair_ac/esppac.cpp:160:32: warning: 'esphome::select::Select::state' is deprecated: Use current_option() instead of .state. Will be removed in 2026.5.0 [-Wdeprecated-declarations]
  160 |         this->display_select_->state != this->display_state_)
      |                                ^~~~~
src/esphome/components/select/select.h:39:15: note: declared here
   39 |   std::string state{};
      |               ^~~~~
src/esphome/components/sinclair_ac/esppac.cpp:160:32: warning: 'esphome::select::Select::state' is deprecated: Use current_option() instead of .state. Will be removed in 2026.5.0 [-Wdeprecated-declarations]
  160 |         this->display_select_->state != this->display_state_)
      |                                ^~~~~
src/esphome/components/select/select.h:39:15: note: declared here
   39 |   std::string state{};
      |               ^~~~~
src/esphome/components/sinclair_ac/esppac.cpp: In member function 'void esphome::sinclair_ac::SinclairAC::update_display_unit(const std::string&)':
src/esphome/components/sinclair_ac/esppac.cpp:171:37: warning: 'esphome::select::Select::state' is deprecated: Use current_option() instead of .state. Will be removed in 2026.5.0 [-Wdeprecated-declarations]
  171 |         this->display_unit_select_->state != this->display_unit_state_)
      |                                     ^~~~~
src/esphome/components/select/select.h:39:15: note: declared here
   39 |   std::string state{};
      |               ^~~~~
src/esphome/components/sinclair_ac/esppac.cpp:171:37: warning: 'esphome::select::Select::state' is deprecated: Use current_option() instead of .state. Will be removed in 2026.5.0 [-Wdeprecated-declarations]
  171 |         this->display_unit_select_->state != this->display_unit_state_)
      |                                     ^~~~~
src/esphome/components/select/select.h:39:15: note: declared here
   39 |   std::string state{};
      |               ^~~~~
src/esphome/components/sinclair_ac/esppac.cpp:171:37: warning: 'esphome::select::Select::state' is deprecated: Use current_option() instead of .state. Will be removed in 2026.5.0 [-Wdeprecated-declarations]
  171 |         this->display_unit_select_->state != this->display_unit_state_)
      |                                     ^~~~~
src/esphome/components/select/select.h:39:15: note: declared here
   39 |   std::string state{};
      |               ^~~~~
```

--- a/components/sinclair_ac/esppac.cpp
+++ b/components/sinclair_ac/esppac.cpp
@@ -250,7 +250,11 @@ void SinclairAC::set_current_temperature_sensor(sensor::Sensor *current_temperat
 void SinclairAC::set_vertical_swing_select(select::Select *vertical_swing_select)
 {
     this->vertical_swing_select_ = vertical_swing_select;
-    this->vertical_swing_select_->add_on_state_callback([this](const std::string &value, size_t index) {
+    this->vertical_swing_select_->add_on_state_callback([this](size_t index) {
+        auto selected = this->vertical_swing_select_->at(index);
+        if (!selected.has_value())
+            return;
+        auto &value = selected.value();
         if (value == this->vertical_swing_state_)
             return;
         this->on_vertical_swing_change(value);
@@ -260,7 +264,11 @@ void SinclairAC::set_vertical_swing_select(select::Select *vertical_swing_select
 void SinclairAC::set_horizontal_swing_select(select::Select *horizontal_swing_select)
 {
     this->horizontal_swing_select_ = horizontal_swing_select;
-    this->horizontal_swing_select_->add_on_state_callback([this](const std::string &value, size_t index) {
+    this->horizontal_swing_select_->add_on_state_callback([this](size_t index) {
+        auto selected = this->horizontal_swing_select_->at(index);
+        if (!selected.has_value())
+            return;
+        auto &value = selected.value();
         if (value == this->horizontal_swing_state_)
             return;
         this->on_horizontal_swing_change(value);
@@ -270,7 +278,11 @@ void SinclairAC::set_horizontal_swing_select(select::Select *horizontal_swing_se
 void SinclairAC::set_display_select(select::Select *display_select)
 {
     this->display_select_ = display_select;
-    this->display_select_->add_on_state_callback([this](const std::string &value, size_t index) {
+    this->display_select_->add_on_state_callback([this](size_t index) {
+        auto selected = this->display_select_->at(index);
+        if (!selected.has_value())
+            return;
+        auto &value = selected.value();
         if (value == this->display_state_)
             return;
         this->on_display_change(value);
@@ -280,7 +292,11 @@ void SinclairAC::set_display_select(select::Select *display_select)
 void SinclairAC::set_display_unit_select(select::Select *display_unit_select)
 {
     this->display_unit_select_ = display_unit_select;
-    this->display_unit_select_->add_on_state_callback([this](const std::string &value, size_t index) {
+    this->display_unit_select_->add_on_state_callback([this](size_t index) {
+        auto selected = this->display_unit_select_->at(index);
+        if (!selected.has_value())
+            return;
+        auto &value = selected.value();
         if (value == this->display_unit_state_)
             return;
         this->on_display_unit_change(value);

--- a/components/sinclair_ac/esppac.cpp
+++ b/components/sinclair_ac/esppac.cpp
@@ -135,7 +135,7 @@ void SinclairAC::update_swing_horizontal(const std::string &swing)
     this->horizontal_swing_state_ = swing;
 
     if (this->horizontal_swing_select_ != nullptr &&
-        this->horizontal_swing_select_->state != this->horizontal_swing_state_)
+        this->horizontal_swing_select_->current_option().str() != this->horizontal_swing_state_)
     {
         this->horizontal_swing_select_->publish_state(this->horizontal_swing_state_);
     }
@@ -146,7 +146,7 @@ void SinclairAC::update_swing_vertical(const std::string &swing)
     this->vertical_swing_state_ = swing;
 
     if (this->vertical_swing_select_ != nullptr && 
-        this->vertical_swing_select_->state != this->vertical_swing_state_)
+        this->vertical_swing_select_->current_option().str() != this->vertical_swing_state_)
     {
         this->vertical_swing_select_->publish_state(this->vertical_swing_state_);
     }
@@ -157,7 +157,7 @@ void SinclairAC::update_display(const std::string &display)
     this->display_state_ = display;
 
     if (this->display_select_ != nullptr && 
-        this->display_select_->state != this->display_state_)
+        this->display_select_->current_option().str() != this->display_state_)
     {
         this->display_select_->publish_state(this->display_state_);
     }
@@ -168,7 +168,7 @@ void SinclairAC::update_display_unit(const std::string &display_unit)
     this->display_unit_state_ = display_unit;
 
     if (this->display_unit_select_ != nullptr && 
-        this->display_unit_select_->state != this->display_unit_state_)
+        this->display_unit_select_->current_option().str() != this->display_unit_state_)
     {
         this->display_unit_select_->publish_state(this->display_unit_state_);
     }

--- a/components/sinclair_ac/esppac_cnt.cpp
+++ b/components/sinclair_ac/esppac_cnt.cpp
@@ -236,7 +236,7 @@ void SinclairACCNT::send_packet()
     bool    fanTurbo  = false;
     if (this->has_custom_fan_mode())
     {
-        const char* custom_fan_mode = this->get_custom_fan_mode();
+        const char* custom_fan_mode = this->get_custom_fan_mode().c_str();
 
         if (strcmp(custom_fan_mode, fan_modes::FAN_AUTO) == 0)
         {
@@ -616,7 +616,7 @@ bool SinclairACCNT::processUnitReport()
     const char* newFanMode = determine_fan_mode();
     if (this->has_custom_fan_mode())
     {
-        if (strcmp(this->get_custom_fan_mode(), newFanMode) != 0) hasChanged = true;
+        if (strcmp(this->get_custom_fan_mode().c_str(), newFanMode) != 0) hasChanged = true;
     }
     else
     {

--- a/examples/ac-sinclair-main.yaml
+++ b/examples/ac-sinclair-main.yaml
@@ -51,7 +51,7 @@ climate:
     xfan_switch:
       name: ${devicename} X-fan
     save_switch:
-      name: ${devicename} Save/8 Heat
+      name: ${devicename} Save|8 Heat
 
 wifi:
   ssid: !secret wifi_ssid

